src/components/Nav/Nav.tsx

```diff
@@ -111,32 +161,37 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
 
   private _renderGroup(group: INavLinkGroup, groupIndex: number): React.ReactElement<{}> {
     const isGroupExpanded: boolean = this.state.isGroupExpanded[groupIndex] !== false;
-    const hasGroupButton: boolean = !!(group.name);
 
     return (
       <div key={ groupIndex } className={ css('ms-Nav-group', { 'is-expanded' : isGroupExpanded }) }>
-        { (hasGroupButton ?
+        { (group.name ?
         <button
-          className='ms-Nav-groupButton'
+          className='ms-Nav-chevronButton ms-Nav-chevronButton--group'
           onClick={ this._onGroupHeaderClicked.bind(this, groupIndex) }
         >
-          <i className={ css('ms-Nav-groupChevron', 'ms-Icon', 'ms-Icon--ChevronDown') }></i>
+          <i className={ css('ms-Nav-chevron', 'ms-Icon', 'ms-Icon--chevronDown') }></i>
           { group.name }
         </button> : null)
         }
-
         <div className={ css('ms-Nav-groupContent', 'ms-u-slideDownIn20') }>
-          { this._renderLinks(group.links, 0 /* nestingLevel */, hasGroupButton) }
+          { this._renderLinks(group.links, 0 /* nestingLevel */) }
         </div>
       </div>
     );
   }
 
   private _onGroupHeaderClicked(groupIndex: number, ev: React.MouseEvent): void {
-    const currentState: boolean = this.state.isGroupExpanded[groupIndex] !== false;
+    let isGroupExpanded: boolean[] = this.state.isGroupExpanded;
+    isGroupExpanded[groupIndex] = !isGroupExpanded[groupIndex];
+    this.setState({ isGroupExpanded: isGroupExpanded });
```

Comment:
Did you know about shorthand properties in JSON?

any time you say:
```
{ foo: foo }
```
you can reduce it to:
```
{ foo }
```